packages/nodes-base/nodes/Splunk/Splunk.node.ts

```diff
@@ -454,18 +457,30 @@ export class Splunk implements INodeType {
 				}
 			} catch (error) {
 				if (this.continueOnFail()) {
-					returnData.push({ error: error.cause.error });
+					returnData.push({ json: { error: error.cause.error }, pairedItem: { item: i } });
 					continue;
 				}
 
-				throw error;
+				if (error instanceof NodeApiError) {
+					set(error, 'context.itemIndex', i);
+				}
+
+				if (error instanceof NodeOperationError && error?.context?.itemIndex === undefined) {
+					set(error, 'context.itemIndex', i);
+				}
+
+				throw new NodeOperationError(this.getNode(), error, { itemIndex: i });
 			}
 
-			Array.isArray(responseData)
-				? returnData.push(...(responseData as IDataObject[]))
-				: returnData.push(responseData as IDataObject);
+			if (Array.isArray(responseData)) {
+				for (const item of responseData) {
+					returnData.push({ json: item, pairedItem: { item: i } });
+				}
+			} else {
+				returnData.push({ json: responseData, pairedItem: { item: i } });
+			}
 		}
 
-		return [this.helpers.returnJsonArray(returnData)];
+		return [responseData];
```

Comment:
I think this should be `returnData`

```suggestion
		return [returnData];
```

Reply:
thanks!